### PR TITLE
add starknet-dss test helpers ✨

### PR DESCRIPTION
**Pull request by EvolveArt**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,49 +1,49 @@
...
+		"lint:fix": "yarn lint --fix",
+		"test": "mocha --exit --recursive 'test/**/*.test.ts'",
+		"test:wait": "wget -q --retry-connrefused --delete-after http://0.0.0.0:5050/fork_status && yarn test",
+		"test:ci": "concurrently -k 'yarn devnet' 'yarn test:wait; docker stop devnet'"
+	},
+	"devDependencies": {
+		"@nomicfoundation/hardhat-network-helpers": "^1.0.7",
+		"@nomiclabs/hardhat-ethers": "^2.2.1",
+		"@shardlabs/starknet-hardhat-plugin": "^0.7.1",
+		"@types/chai": "^4.3.4",
```
Comment (maciejka) [inline package.json]: Do we want both chai and earljs? I don't have any preferences

File: test/helpers/starknet/utils.ts [hunk 2]
```diff
@@ -3,14 +3,22 @@ import { HardhatRuntimeEnvironment } from "hardhat/types";
...
-    hre,
-    abiPath: path.join(process.env.PWD!, abiPath),
-    metadataPath: "", //ignored
-  });
-  return factory.getContractAt(address);
+	const factory = new StarknetContractFactory({
+		hre,
+		abiPath: path.join(
+			process.env.PWD!,
+			"/starknet-artifacts/contracts/starknet/starknet-dss/",
```
Comment (maciejka) [inline test/helpers/starknet/utils.ts:14]: starknet-dss is not the only Cairo dependency

File: test/setup.ts [hunk 3]
```diff
@@ -1,123 +1,145 @@
...
+
+	const dss: DssInstance = {
+		vat: await getVat("0x35D1b3F3D7966A1DFe207aa4514C12a259A0492B"),
+		jug: await getJug("0x19c0976f590D67707E62397C87829d896Dc0f1F1"),
+		cure: await getCure("0x0085c9feAb2335447E1F4DC9bf3593a8e28bdfc7"),
+		vow: await getVow("0xA950524441892A31ebddF91d3cEEFa04Bf454466"),
+	};
+
+	// This Sets `snPredeployedAccounts[1]` as the global active account
+	const sn_dss: starknetDss.SNDssInstance = await starknetDss.deploy(
```
Comment (maciejka) [inline test/setup.ts]: Why snake case?

File: test/starknet-dss/starknetDss.ts [hunk 4]
```diff
@@ -1,19 +1,251 @@
...
+	};
 }
 
-// TODO: add more contracts
 // TODO: add init script
 // Based on: https://github.com/makerdao/dss-bridge/blob/4cfc84761b4bfeae747af14d3a2545377dd3304a/src/deploy/XDomainDss.sol
+export async function getSNDaiJoin(address: string): Promise<SNDaiJoin> {
+	const daiJoin = await getL2ContractAt(
+		hre,
+		"dai_join.cairo/dai_join_abi.json",
```
Comment (maciejka) [inline test/starknet-dss/starknetDss.ts:188]: We should ask Shardlabs to accept abi object instead of a path.

File: package.json (hunk 1 above)
Comment (EvolveArt) [reply package.json]: earljs is good

File: test/setup.ts (hunk 3 above)
Comment (EvolveArt) [reply test/setup.ts]: do we prefer SNDss here or snDSS tho ?

File: test/starknet-dss/starknetDss.ts (hunk 4 above)
Comment (EvolveArt) [reply test/starknet-dss/starknetDss.ts:188]: definitelt would make more sense

File: test/setup.ts (hunk 3 above)
Comment (maciejka) [reply test/setup.ts]: snDss

**Review by maciejka — APPROVED** [review]

**Comment by maciejka** [comment]
Merged into starknet-dss-teleport.

**Pull request closed without merging**